test/dd/test_dd_functionality.cpp

```diff
@@ -121,7 +121,7 @@ TEST_P(DDFunctionality, controlledStandardOpBuildInverseBuild) {
   using namespace qc::literals;
   auto gate = static_cast<qc::OpType>(GetParam());
 
-  qc::StandardOperation op;
+  qc::StandardOperation op, op_neg;
```

Comment:
I'd slightly prefer to refactor this into two separate tests. Let's keep the original test and extract the negative control one into a separate function.
Then, you can also directly test the single-target gates with negative controls, which should yield even better coverage.

Reply:
Done 👍🏼